client/scripts/controllers/chain/ethereum/marlin/api.ts

```diff
@@ -0,0 +1,77 @@
+import { Web3Provider, AsyncSendable, JsonRpcSigner, Provider } from 'ethers/providers';
+import { ethers } from 'ethers';
+
+import { Erc20 } from 'Erc20';
+import { Erc20Factory } from 'Erc20Factory';
+
+import { MPond } from 'MPond';
+import { MPondFactory } from 'MPondFactory';
+import { GovernorAlpha } from 'GovernorAlpha';
+import { GovernorAlphaFactory } from 'GovernorAlphaFactory';
+import { BigNumber } from 'ethers/utils';
+
+export default class MarlinAPI {
+  public readonly gasLimit: number = 3000000;
+
+  private _userAddress: string;
+  private _MPondAddress: string;
+  private _GovernorAlphaAddress: string;
+  private _MPondContract: MPond;
+  private _GovernorAlphaContract: GovernorAlpha;
+  private _Provider: Web3Provider;
+  private _Signer: JsonRpcSigner;
+  private _tokenContract: Erc20;
+  private _userMPond: BigNumber;
+  private _Symbol: string;
+  private _Bridge: string;
+
+  public get userAddress() { return this._userAddress; }
+  public get mPondAddress() { return this._MPondAddress; }
+  public get governorAlphaAddress() { return this._GovernorAlphaAddress; }
+  public get mPondContract(): MPond { return this._MPondContract; }
+  public get governorAlphaContract(): GovernorAlpha { return this._GovernorAlphaContract; }
+
+  public get Provider(): Web3Provider { return this._Provider; }
+  public get Signer(): JsonRpcSigner { return this._Signer; }
+  public get tokenContract() { return this._tokenContract; }
+  public get userMPond(): number { return this._userMPond.toNumber(); }
+  public get symbol(): string { return this._Symbol; }
+  public get bridge(): string { return this._Bridge; }
+
+  constructor(
+    mPondAddress: string,
+    governorAlphaAddress: string,
+    web3Provider: AsyncSendable,
+    userAddress?: string,
+  ) {
+    if (userAddress) {
+      this._userAddress = userAddress.toLowerCase();
+      this._Signer = this._Provider.getSigner(userAddress);
+    }
+
+    this._Provider = new ethers.providers.Web3Provider(web3Provider);
```

Comment:
We should check if this is a web wallet and if so, directly initialize the Signer. As is, we need to change the Metamask and change back to account to initialize the Signer.

Reply:
We don't check this in the marlin api, but we pass it in to the marlin api constructor from the chain-initialization step prior. We just handle it as `web3Provider` of type `asyncSendable`.

Aside, I changed the order around, moving line 52 to line 47, because we need the Provider prior to call you added to get the signer.